src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/UploadContestedCaseDocumentsService.java

```diff
@@ -0,0 +1,113 @@
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.ContestedUploadedDocumentData;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APPLICANT_CORRESPONDENCE_COLLECTION;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APPLICANT_EVIDENCE_COLLECTION;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APPLICANT_FR_FORM_COLLECTION;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APPLICANT_TRIAL_BUNDLE_COLLECTION;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.CONTESTED_UPLOADED_DOCUMENTS;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class UploadContestedCaseDocumentsService {
+
+    private static final String APPLICANT = "applicant";
+
+    private final ObjectMapper mapper;
+
+    public Map<String, Object> filterDocumentsToRelevantParty(Map<String, Object> caseData) throws Exception {
+
+        List<ContestedUploadedDocumentData> uploadedDocuments = getDocumentCollection(caseData, CONTESTED_UPLOADED_DOCUMENTS);
+
+        //we need to build ApplicantUploadedDocumentData, and add the "Applicant" marked doc a collection
+        List<ContestedUploadedDocumentData> correspondenceFiltered = uploadedDocuments.stream()
+            .filter(d -> d.getUploadedCaseDocument().getCaseDocumentParty() != null
+                && d.getUploadedCaseDocument().getCaseDocumentParty().equals(APPLICANT))
+            .filter(d -> d.getUploadedCaseDocument().getCaseDocumentType().equals("Letter from Applicant"))
+            .collect(Collectors.toList());
+        List<ContestedUploadedDocumentData> applicantCorrespondenceCollection = getDocumentCollection(caseData, APPLICANT_CORRESPONDENCE_COLLECTION);
+        applicantCorrespondenceCollection.addAll(correspondenceFiltered);
+        log.info("Adding item: {}, to Applicant Correspondence Collection", correspondenceFiltered);
+
+        List<ContestedUploadedDocumentData> frFormsFiltered = uploadedDocuments.stream()
+            .filter(d -> d.getUploadedCaseDocument().getCaseDocumentParty() != null
+                && d.getUploadedCaseDocument().getCaseDocumentParty().equals(APPLICANT))
+            .filter(d -> d.getUploadedCaseDocument().getCaseDocumentType().equals("Form B")
+                    || d.getUploadedCaseDocument().getCaseDocumentType().equals("Applicant - Form E")
+                    || d.getUploadedCaseDocument().getCaseDocumentType().equals("Form F")
+                    || d.getUploadedCaseDocument().getCaseDocumentType().equals("Form G")
+                    || d.getUploadedCaseDocument().getCaseDocumentType().equals("Form H"))
```

Comment:
filter might be more readable splitting the conditions out to separate methods ie
` .filter(d -> isTypeValid(d.getUploadedCaseDocument().getCaseDocumentType())


private boolean isTypeValid(String caseType) {
return caseType.equals("Form B") || caseType.equals....
}
`

Reply:
Cheers for the suggestion, it was previously in a map. Refactored it back 👍